app/controllers/benefit_overrides_controller.rb

```diff
@@ -29,10 +29,19 @@ def application
   end
 
   def benefit_override
-    @benefit_override ||= BenefitOverride.find_or_create_by(application_id: application.id)
+    @benefit_override ||= BenefitOverride.find_or_initialize_by(application: application)
   end
 
   def allowed_params
-    params.require(:benefit_override).permit(*Forms::BenefitsEvidence.permitted_attributes)
+    params.require(:benefit_override).permit(*Forms::BenefitsEvidence.permitted_attributes.keys)
+  end
+
+  helper_method def error_message_partial
+    @error_message_partial ||= case application.last_benefit_check.dwp_result.try(:downcase)
+                               when nil, 'undetermined'
+                                 'missing_details'
+                               when 'server unavailable', 'unspecified error'
+                                 'technical_error'
+                               end
```

Comment:
I personally find the `case` version more readable and it's more efficient. @colinbruce which one do you prefer?